### PR TITLE
Export-Import Wallet using QR code

### LINKED ISSUE #4320: Import wallet on a new device by scanning a QR code

### PR DESCRIPTION
**Pull request by JDonadio**
* Add tabs in export view (File - QR)
* Add advanced options to display the priv key inclusion
* Add QR scan button at import view (mnemonics)
* Prevent to show scanned data (lock icon)
* Improve error messages
* Warning message if has passphrase

Closes #4320

### CODE REVIEW

**Comment by matiu** [comment]
epic!

**Comment by matiu** [comment]
please link the related tickets

File: public/views/export.html [hunk 1]
```diff
@@ -5,89 +5,118 @@
...
 
-            <label for="password" translate>Repeat the password</label>
-            <div class="input">
-              <input type="password" class="form-control" placeholder="{{'Repeat password'|translate}}" name="password" ng-model="repeatpassword">
-            </div>
-          </div>
-        </div>
-      </form>
+      <div class="text-center size-18 m10t" ng-style="{color: index.backgroundColor}">
+        <span translate>Exporting Walelt: </span>
```
Comment (cmgustavo) [inline public/views/export.html]: Wording error: `Walelt`

File: public/views/import.html [hunk 2]
```diff
@@ -23,118 +23,116 @@
...
-              <div class="m15l">
-                <span translate>NOTE: To import a wallet from a 3rd party software, please go to Add Wallet &gt; Create Wallet, and specify the Recovery Phrase there.</span><br>
-              </div>
+        <div class="box-notification m20b" ng-show="importErr">
+          <div ng-show="importErr" class="text-warning">
+            <div class="m10 text-bold" translate>Could not access the wallet at the server. Please check:</div>
+            <ul class="size-12">
+              <li translate>The password of the recovery phrase (if set)</li>
+              <li translate>The derivation path</li>
+              <li translate>The wallet service URL
```
Comment (cmgustavo) [inline public/views/import.html]: Close `</li>` if there is a `translate` attribute before

File: src/js/controllers/backup.js [hunk 3]
```diff
@@ -23,8 +23,7 @@ angular.module('copayApp.controllers').controller('backupController',
 
         handleEncryptedWallet(fc, function(err) {
           if (err) {
-            $scope.error = bwsError.msg(err, gettext('Could not decrypt'));
-            $log.warn('Error decrypting credentials:', $scope.error);
```
Comment (cmgustavo) [inline src/js/controllers/backup.js]: Keep `$log`

File: public/views/export.html [hunk 4]
```diff
@@ -5,89 +5,118 @@
...
+        <span>{{index.walletName}}</span>
+      </div>
+      <div class="text-center size-12 m10" translate>From/In the destination device, go to Add wallet > Import wallet and scan this QR code</div>
```
Comment (isocolsky) [inline public/views/export.html]: Should be either "From" or "In".

File: public/views/export.html [hunk 5]
```diff
@@ -5,89 +5,118 @@
...
 
 <div class="content preferences" ng-controller="exportController" ng-init="init(index.prevState)">
-  <h4></h4>
-  <div ng-show="!backupWalletPlainText">
-      <div class="text-warning size-14 m20b" ng-show="error">
-        <i class="fi-alert size-12"></i>
-        <span translate>Failed to export</span>
+  <div class="create-tab small-only-text-center">
+    <div class="row">
+      <div class="tab-container small-6 medium-3 large-2"
```
Comment (cmgustavo) [inline public/views/export.html]: missing `columns`

File: src/js/services/onGoingProcess.js [hunk 6]
```diff
@@ -28,6 +28,7 @@ angular.module('copayApp.services').factory('ongoingProcess', function($log, $ti
     'importingWallet': gettext('Importing Wallet...'),
     'sweepingWallet': gettext('Sweeping Wallet...'),
     'deletingWallet': gettext('Deleting Wallet...'),
+    'processingCode': gettext('Processing code...'),
```
Comment (isocolsky) [inline src/js/services/onGoingProcess.js]: 'Extracting wallet information...'

File: src/js/controllers/import.js [hunk 7]
```diff
@@ -1,58 +1,124 @@
...
-        $scope.seedSource = self.seedOptions[0];
+        $scope.seedSource = $scope.seedOptions[0];
+      }
+    };
+
+    $scope.processCode = function(code) {
+      $scope.fromQR = null;
+      $scope.importErr = false;
+      $scope.error = null;
+      ongoingProcess.set('processingCode', true);
```
Comment (cmgustavo) [inline src/js/controllers/import.js]: It doesn't work on iOS

**Comment by JDonadio** [comment]
Updated. Pending to import when the privkey is not included in the code. Thanks guys for the comments!

**Comment by gabrielbazan7** [comment]
ACK!

**Comment by cmgustavo** [comment]
LGTM!!

**Comment by gabrielbazan7** [comment]
ACK !

**Comment by isocolsky** [comment]
Great Job!! ACK

**Pull request merged**